File: frontend/src/types/account.ts
export const ACCOUNT_TYPES = ['checking', 'savings', 'credit_card', 'cash', 'other'] as const;
export type AccountType = typeof ACCOUNT_TYPES[number];

export interface Account {
  id: string;
  user_id: string;
  name: string;
  type: AccountType;
  currency: string;
  initial_balance: string;
  is_active: boolean;
  balance: string | null;
  created_at: string;
  updated_at: string;
}

export interface AccountCreate {
  name: string;
  type: AccountType;
  currency: string;
  initial_balance?: string;
  is_active?: boolean;
}

export interface AccountUpdate {
  name?: string;
  type?: AccountType;
  is_active?: boolean;
}

File: backend/app/schemas/account.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel

ACCOUNT_TYPES = ["checking", "savings", "credit_card", "cash", "other"]


class AccountCreate(BaseModel):
    name: str
    type: str
    currency: str
    initial_balance: Decimal = Decimal("0")
    is_active: bool = True


class AccountUpdate(BaseModel):
    name: str | None = None
    type: str | None = None
    is_active: bool | None = None


class AccountResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    name: str
    type: str
    currency: str
    initial_balance: Decimal
    is_active: bool
    balance: Decimal | None = None  # Computed field
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
